src/types/service-worker.d.ts

```diff
@@ -0,0 +1,36 @@
+/// <reference lib="webworker" />
+
+// Extend ServiceWorkerRegistration with sync APIs
+interface ServiceWorkerRegistration {
+  sync?: SyncManager;
+  periodicSync?: PeriodicSyncManager;
```

Comment:
Shouldn't these come from an official source of web API types? Do we need to update to a newer version of the type definitions?